libdrgn/program.c

```diff
@@ -86,6 +87,37 @@ void drgn_program_init(struct drgn_program *prog,
 	drgn_object_init(&prog->vmemmap, prog);
 }
 
+static int drgn_program_segments_deinit(struct drgn_program *prog)
+{
+	GElf_Phdr phdr_mem, *phdr;
+	size_t phnum, i, j;
+	uint64_t file_size;
+	void *map_address;
+
+	if (elf_getphdrnum(prog->core, &phnum) != 0)
```

Comment:
Rather than digging through the core file again, let's just save `num_file_segments` in `struct drgn_program`.